File: Java/Warmup-1/frontBack.java
package test;

import static org.junit.Assert.*;

import org.junit.Test;

/*
 * 
Given a string, return a new string where the first and last chars have been exchanged.


frontBack("code") → "eodc"
frontBack("a") → "a"
frontBack("ab") → "ba"
 */

public class frontBack {

	private static int i = 0;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("frontBack(\"code\") → \"eodc\"\n");
		assertEquals("frontBack(\"code\") → \"eodc\"", "eodc", frontBack("code"));
		i++;
	}
	
	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("frontBack(\"a\") → \"a\"\n");
		assertEquals("frontBack(\"a\") → \"a\"", "a", frontBack("a"));
		i++;
	}

	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("frontBack(\"ab\") → \"ba\"\n");
		assertEquals("frontBack(\"ab\") → \"ba\"", "ba", frontBack("ab"));
		i++;
	}
	
	public String frontBack(String str) {
		
		if(str.length() <2) {
			return str;
		}
		
		else {
			return str.charAt(str.length()-1) + str.substring(1, str.length()-1) + str.charAt(0);
		}
		
	}

	
}

File: Python/Warmup-1/front_back.py
import unittest

'''


Given a string, return a new string where the first and last chars have been exchanged.


front_back('code')  -> 'eodc'
front_back('a')     -> 'a'
front_back('ab')    -> 'bafrom symbol import import_as_name
'

'''

def front_back(str):

    if len(str) == 1:
        return str
 
    elif len(str) == 2:
        return str[1]+str[0]
  
    elif len(str) > 2:  
        return str[len(str)-1] + str[1:(len(str)-1)] + str[0]  

    return '' 
 
 
 
class Test_front_back(unittest.TestCase):

    def test_1(self):
        self.assertEqual(front_back('code'), 'eodc')

    def test_2(self):
        self.assertEqual(front_back('a'), 'a')

    def test_3(self):
        self.assertEqual(front_back('ab'), 'ba')


if __name__ == '__main__':
    unittest.main() 
    
  